src/tokenize.cpp

```diff
@@ -1913,6 +1935,21 @@ void tokenize(const deque<int> &data, chunk_t *ref)
             cpd.preproc_ncnl_count++;
          }
 
+         /* Figure out the type of pragma for enable/disable indentation */
+         if (cpd.in_preproc == CT_PP_PRAGMA)
+         {
+            if (memcmp(pc->text(),"asm",3) == 0)
```

Comment:
Sorry but I don't see the util function you are referring to.

Not sure what you meant by "here as well". The first checks (starting line 1913) are for detecting directive `#asm`. The second checks (starting  like 1936) are for detecting directive `#pragma asm`.

Now, maybe you ask why I test for the end of section (i.e. directives `#endasm` and `#pragma endasm`). Reading the code again, maybe this is in fact not needed, since only effect is to reset the flag `cpd.unc_off`, which was done already.